components/ai_chat/core/browser/engine/conversation_api_client.cc

```diff
@@ -0,0 +1,299 @@
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("ai_chat", R"(
+      semantics {
+        sender: "AI Chat"
+        description:
+          "This is used to communicate with Brave's AI Conversation API"
+          "on behalf of the user interacting with different browser AI"
+          "features."
+        trigger:
+          "Triggered by user interactions such as submitting an AI Chat"
+          "conversation message, or requesting a text rewrite."
+        data:
+          "Conversational messages input by the user as well as associated"
+          "content or user text to be rewritten. Can contain PII."
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+base::Value::List ConversationEventsToList(
+    const std::vector<ConversationEvent>& conversation) {
+  base::Value::List events;
+  for (const auto& event : conversation) {
+    base::Value::Dict event_dict;
+    // Set role
+    switch (event.role) {
+      case mojom::CharacterType::HUMAN:
+        event_dict.Set("role", "user");
+        break;
+      case mojom::CharacterType::ASSISTANT:
+        event_dict.Set("role", "assistant");
+        break;
+      default:
+        NOTREACHED();
+        break;
+    }
+
+    switch (event.type) {
+      case ConversationEventType::ContextURL:
+        event_dict.Set("type", "contextURL");
+        break;
+      case ConversationEventType::UserText:
+        event_dict.Set("type", "userText");
+        break;
+      case ConversationEventType::PageText:
+        event_dict.Set("type", "pageText");
+        break;
+      case ConversationEventType::PageExcerpt:
+        event_dict.Set("type", "pageExcerpt");
+        break;
+      case ConversationEventType::VideoTranscriptXML:
+        event_dict.Set("type", "videoTranscriptXML");
+        break;
+      case ConversationEventType::VideoTranscriptVTT:
+        event_dict.Set("type", "videoTranscriptVTT");
+        break;
+      case ConversationEventType::ChatMessage:
+        event_dict.Set("type", "chatMessage");
+        break;
+      case ConversationEventType::RequestRewrite:
+        event_dict.Set("type", "requestRewrite");
+        break;
+      case ConversationEventType::RequestSummary:
+        event_dict.Set("type", "requestSummary");
+        break;
+      case ConversationEventType::RequestSuggestedActions:
+        event_dict.Set("type", "requestSuggestedActions");
+        break;
+      case ConversationEventType::SuggestedActions:
+        event_dict.Set("type", "suggestedActions");
+        break;
+      default:
+        NOTREACHED();
+        break;
+    }
+    event_dict.Set("content", event.content);
+    events.Append(std::move(event_dict));
+  }
+  return events;
+}
+
+GURL GetEndpointUrl(bool premium, const std::string& path) {
+  DCHECK(!path.starts_with("/"));
```

Comment:
https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/checks.md#invariant_verification-mechanisms

Reply:
Sure this can be a `CHECK`, but I think it needs to be there as the point of it is to make sure something that's not obvious from the function params (`path` shouldn't start with `/`). If this were a public function I'd consider adding a `base::TrimString(path, "/", base::TRIM_LEADING)` at the start of the function body. What do you think?

Reply:
Let's do CHECK for the two existing DCHECK in this function then, and we should make sure URL being used is always covered by unit tests so we know CHECK aren't hit.